tockloader-lib/src/attributes/decode.rs

```diff
@@ -17,7 +18,15 @@ impl DecodedAttribute {
     }
 }
 
-// TODO: explain what is happening here
+// Function used to decode 64 byte chunks from the attributes region at the 0x600-0x9FF range
+// The first 8 bytes represent the key of the attribute,
+// while the rest represent the stored data.
+// The key is utf-8 decoded and converted into a String.
+// Then we check the vlen value at the 9th byte,
+// representing the bytewise length of the value from the attribute.
+// Given that the chunks contain 64 bytes the value can not be larger than 55.
+// We return None if the data is larger than the possible length or if the data does not exist for that particular attribute.
+// Afterwards we decode the bytes of the value and turn them into a String.
 pub(crate) fn decode_attribute(step: &[u8]) -> Option<DecodedAttribute> {
     let raw_key = &step[0..8];
 
```

Comment:
```suggestion
/// All attributes are stored in 64 byte chunks, one after another in the 0x600-0x9FF range. This
/// function is used to interpret the raw data from within that region.
/// 
/// An attribute is composed of:
/// - utf-8 encoded key (bytes 1 to 8)
/// - value length (byte 9)
/// - utf-8 encoded value (bytes 10 to 10 + length)
/// - padding (remaining bytes)
/// 
/// # Arguments
/// - `step: &[u8]` - the byte slice to read the next attribute from.
/// 
/// # Return value
/// This function returns None if the byte slice is too short, the `value length` attribute is too
/// large (maximum valid value is 55) or "if the data does not exist for that particular attribute"
/// (TODO: what does this mean?)
pub(crate) fn decode_attribute(step: &[u8]) -> Option<DecodedAttribute> {
    // TODO: return result after we tackle error-handling
    // TODO: verify `step` length. This will panic otherwise
    let raw_key = &step[0..8];
```